nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ReplaceText.java

```diff
@@ -386,31 +385,28 @@ public void process(final InputStream in, final OutputStream out) throws IOExcep
                     }
                 });
             } else {
-                flowFile = session.write(flowFile, new StreamReplaceCallback(this, charset, maxBufferSize, context, flowFile,  null));
-            }
+                flowFile = session.write(flowFile, new StreamReplaceCallback(charset, maxBufferSize, context.getProperty(LINE_BY_LINE_EVALUATION_MODE).getValue(),
+                    ((bw, oneLine) -> {
+                        // We need to determine what line ending was used and use that after our replacement value.
+                        lineEndingBuilder.setLength(0);
+                        for (int i = oneLine.length() - 1; i >= 0; i--) {
+                            final char c = oneLine.charAt(i);
+                            if (c == '\r' || c == '\n') {
+                                lineEndingBuilder.append(c);
+                            } else {
+                                break;
+                            }
+                        }
 
-            return flowFile;
-        }
+                        bw.write(replacementValue);
 
-        public void replaceInLine(BufferedWriter bw, String oneLine, @Nullable Matcher matcher, @Nullable Pattern searchPattern, ProcessContext context, FlowFile flowFile) throws IOException {
-            final String replacementValue = context.getProperty(REPLACEMENT_VALUE).evaluateAttributeExpressions(flowFile).getValue();
```

Comment:
By using functional approach, we can use closure mechanism, that allows the function body in the lambda block refers preceding final variables such as `lineEndingBuilder` and `replacementValue`.

Reply:
I learnt closure mechanism from this PR. Thanks. Couldn't agree less.